src/dysh/fits/sdfitsload.py

```diff
@@ -420,6 +420,33 @@ def getspec(self, i, bintable=0, observer_location=None):
         restfreq = rfq.to("Hz").value
         meta["RESTFRQ"] = restfreq  # WCS wants no E
 
+        # @todo   could we safely store it in meta['BUNIT']
+        #  for now, loop over the binheader keywords to find the matching TUNITxx that belongs to TTYPExx='DATA'
+        #  if BUNIT was also found, a comparison is made, but TUNIT wins
+        if "BUNIT" in meta:
+            bunit = meta["BUNIT"]
+        else:
+            bunit = None
+            h = self.binheader()[0]
+            for k, v, c in h.cards:
+                if k == "BUNIT":
+                    bunit = v
+            ukey = None
+            for k, v, c in h.cards:  # loop over the (key,val,comment) for all cards in the header
+                if v == "DATA":
+                    ukey = "TUNIT" + k[5:]
+                    break
+            for k, v, c in h.cards:
+                if k == ukey:
```

Comment:
What happens here when ukey is None?

Reply:
it's fine (should be rare that TUNITxx is absent, but it would fall back to the "ct" unit, unless BUNIT in the header, then it will carry that across

Reply:
Seems like you could skip the loop if ukey is None